src/bin/dtnd.rs

```diff
@@ -301,7 +301,7 @@ Tag 255 takes 5 arguments and is interpreted as address. Usage: -S 255:'Samplest
     }
     if let Some(destinations) = matches.values_of("discoverydestination") {
         for destination in destinations {
-            cfg.add_destination(String::from(destination))
+            cfg.add_destination(&mut String::from(destination))
```

Comment:
is it really necessary to pass a mutable string here? does not seem very clean.
From an outside perspective, it is not clear what is happening in add_destination and why something is mutated in there.